src/server/verification/processor/provider/ZoomProvider.js

```diff
@@ -69,44 +48,53 @@ class ZoomProvider implements IEnrollmentProvider {
     await notifyProcessor({ isDuplicate })
 
     if (duplicate) {
-      const duplicateFoundMessage = `Duplicate with identifier '${duplicate.enrollmentIdentifier}' found.`
+      const duplicateFoundMessage = `Duplicate exists for FaceMap you're trying to enroll.`
 
       // if duplicate found - throwing corresponding error
-      throwCustomException(duplicateFoundMessage, { isDuplicate, ...duplicate })
+      throwCustomException(duplicateFoundMessage, { isDuplicate, ...searchResponse })
     }
 
     let enrollmentResult
     let alreadyEnrolled = false
 
-    // 3. performing enroll
+    // 2. performing enroll
     try {
+      // returning last respose
       enrollmentResult = await api.submitEnrollment({ ...payload, enrollmentIdentifier })
     } catch (exception) {
       const { response, message } = exception
 
-      // rethrowing unexpected errors (e.g. no conneciton or service error)
+      // if exception has no response (e.g. no conneciton or service error)
+      // just rethrowing it and stopping enrollment
       if (!response) {
         throw exception
       }
 
-      // if exception isn't related to the case when
-      // facemap is just already enrolled
+      // if exception has response checking
+      // is subCode non-equals to 'nameCollision'
+      // that we have some enrollment error
+      // (e.g. liveness wasn't passsed, glasses detected, poor quality)
       if ('nameCollision' !== response.subCode) {
         const isEnrolled = false
+        const isLive = api.checkLivenessStatus(response)
```

Comment:
does it return liveness in case of collision?

Reply:
Have agreed with the following:

> Hadar R  11:45 AM
> currently if already enroll just approve.
> Hadar R  11:46 AM
> we will add a call to "authenticate" in the near future. meaning if recordId exists it will verify that incoming facemap can be authenticated as same record already in database